src/backend/aspen/api/utils/fasta_streamer.py

```diff
@@ -61,19 +75,43 @@ async def stream(self) -> AsyncGenerator[str, None]:
                 stripped_sequence: str = sequence.strip("Nn")
                 # use private id if the user has access to it, else public id
                 if sample.submitting_group_id == self.ac.group.id:  # type: ignore
-                    yield self._output_id_line(sample.private_identifier)
+                    yield await self._output_id_line(prefix, sample.private_identifier)
                 else:
-                    yield self._output_id_line(sample.public_identifier)
+                    yield await self._output_id_line(prefix, sample.public_identifier)
                 yield stripped_sequence
                 yield "\n"
 
-    def _output_id_line(self, identifier) -> str:
+    async def get_public_repository_prefix(self):
+        if self.pathogen_slug and self.public_repository_name:
+            # only get the prefix if we have enough information to proceed
+            prefix = (
+                sa.select(PathogenRepoConfig)
+                .join(Pathogen)
+                .join(PublicRepository)
+                .where(
+                    and_(
+                        Pathogen.slug == self.pathogen_slug,
+                        PublicRepository.name == self.public_repository_name,
```

Comment:
Just a headsup that this repository name is *case sensitive* and if we don't find a matching public repository, this endpoint doesn't return an error -- it just doesn't add prefixes to our sample names.